File: go/api/arch/corp_coop.go
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/13
 */

package arch

import (
	"api-demo/base"
	"api-demo/model"
	"encoding/json"
)

// CorpCoopApiClient 角色Api
type CorpCoopApiClient struct {
	*base.ApiClient
}

func (api *CorpCoopApiClient) ValidVersion(version string) bool {
	return map[string]bool{
		"v4": true,
	}[version]
}

func (api *CorpCoopApiClient) DefaultVersion() string {
	return "v4"
}

// DoRequest 发送http请求
func (api *CorpCoopApiClient) DoRequest(option *base.RequestOption) (responseBody []byte, err error) {
	if !api.ValidVersion(api.Version) {
		api.Version = api.DefaultVersion()
	}
	return api.ApiClient.DoRequest(option)
}

// CorpCoopDepartList 列出我连接的企业
func (api *CorpCoopApiClient) CorpCoopDepartList(options map[string]interface{}) (responseBody *model.DeptList, err error) {
	payload, err := json.Marshal(options)
	if err != nil {
		return
	}

	response, err := api.DoRequest(&base.RequestOption{
		Method:  "POST",
		Path:    "corp_coop/guest/department_list",
		Payload: payload,
	})
	if err != nil {
		return
	}

	err = json.Unmarshal(response, &responseBody)
	return
}

// CorpCoopMemberList 列出我连接的企业对接人
func (api *CorpCoopApiClient) CorpCoopMemberList(options map[string]interface{}) (responseBody *model.MemberList, err error) {
	payload, err := json.Marshal(options)
	if err != nil {
		return
	}

	response, err := api.DoRequest(&base.RequestOption{
		Method:  "POST",
		Path:    "corp_coop/guest/member_list",
		Payload: payload,
	})
	if err != nil {
		return
	}

	err = json.Unmarshal(response, &responseBody)
	return
}

// CorpCoopUserInfo 列出我连接的企业对接人详细信息
func (api *CorpCoopApiClient) CorpCoopUserInfo(username string) (responseBody *model.Member, err error) {
	payload, err := json.Marshal(map[string]interface{}{
		"username": username,
	})
	if err != nil {
		return
	}

	response, err := api.DoRequest(&base.RequestOption{
		Method:  "POST",
		Path:    "corp_coop/guest/user_retrieve",
		Payload: payload,
	})
	if err != nil {
		return
	}

	err = json.Unmarshal(response, &responseBody)
	return
}

File: csharp/src/api/arch/CorpCoop.cs
/**
* Copyright (c) 2015-2022, FineX, All Rights Reserved.
* @author Thomas.Zhuang
* @date 2022/06/15
*/
using System.Text.Json;
using Src.Base;

namespace Src.Api.Jdy;

public class CorpCoopApiClient : ApiClient
{
    private readonly static string[] validVersions = { "v4" };
    private readonly static string defaultVersion = "v4";
    public CorpCoopApiClient(string host, string apiKey, string version = "") : base(host, apiKey, version)
    {
    }

    public new async Task<JsonElement?> doRequest(IDictionary<string, object> option)
    {
        if (Array.IndexOf(validVersions, base.version) == -1)
        {
            base.version = defaultVersion;
        }
        return await base.doRequest(option);
    }

    /**
    * 列出我连接的企业
    */
    public async Task<JsonElement?> corpCoopDepartList(IDictionary<string, object> option)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "corp_coop/guest/department_list");
        requestOption.Add("payload", option);
        return await this.doRequest(requestOption);
    }

    /**
    * 列出我连接的企业对接人
    */
    public async Task<JsonElement?> corpCoopMemberList(IDictionary<string, object> option)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "corp_coop/guest/member_list");
        requestOption.Add("payload", option);
        return await this.doRequest(requestOption);
    }

    /**
    * 列出我连接的企业对接人详细信息
    */
    public async Task<JsonElement?> CorpCoopUserInfo(string username)
    {
        IDictionary<string, object> requestOption = new Dictionary<string, object>();
        requestOption.Add("method", "POST");
        requestOption.Add("path", "corp_coop/guest/user_retrieve");
        IDictionary<string, object> payload = new Dictionary<string, object>();
        payload.Add("username", username);
        requestOption.Add("payload", payload);
        return await this.doRequest(requestOption);
    }

}